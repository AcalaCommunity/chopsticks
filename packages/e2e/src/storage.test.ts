import '@polkadot/api-augment'
import { describe, expect, it } from 'vitest'

import { api, delay, dev, env, expectJson, mockCallback, setupApi } from './helper'

setupApi(env.mandala)

describe('storage', () => {
  it('getStorage', async () => {
    expectJson(await api.query.timestamp.now()).toMatchSnapshot()
    expectJson(await api.query.system.account('5F98oWfz2r5rcRVnP9VCndg33DAAsky3iuoBSpaPUbgN9AJn')).toMatchSnapshot()

    const apiAt = await api.at('0x6aa33cc6b9bfa9bbc3144597da05a1dfd3bf2d16fc52f67c1687fc4e7c8ced5f')

    expectJson(await apiAt.query.timestamp.now()).toMatchSnapshot()
    expectJson(await apiAt.query.system.account('5F98oWfz2r5rcRVnP9VCndg33DAAsky3iuoBSpaPUbgN9AJn')).toMatchSnapshot()
  })

  it('getStorageMulti', async () => {
    expectJson(
      await api.query.system.account.multi([
        '5F98oWfz2r5rcRVnP9VCndg33DAAsky3iuoBSpaPUbgN9AJn',
        '5Fe3jZRbKes6aeuQ6HkcTvQeNhkkRPTXBwmNkuAPoimGEv45',
        '5EMjsczLB8VLkyWNp8b9v5bnGcNEgeW2PVdFqNmQFPoB6sxk',
      ])
    ).toMatchSnapshot()

    const apiAt = await api.at('0x6aa33cc6b9bfa9bbc3144597da05a1dfd3bf2d16fc52f67c1687fc4e7c8ced5f')

    expectJson(
      await apiAt.query.system.account.multi([
        '5F98oWfz2r5rcRVnP9VCndg33DAAsky3iuoBSpaPUbgN9AJn',
        '5Fe3jZRbKes6aeuQ6HkcTvQeNhkkRPTXBwmNkuAPoimGEv45',
        '5EMjsczLB8VLkyWNp8b9v5bnGcNEgeW2PVdFqNmQFPoB6sxk',
      ])
    ).toMatchSnapshot()
  })

  it('getKeysPaged', async () => {
    const entries = await api.query.system.account.entriesPaged({ args: [], pageSize: 10 })
    expect(entries).toMatchSnapshot()

    const entries2 = await api.query.system.account.entriesPaged({
      args: [],
      pageSize: 10,
      startKey: entries[entries.length - 1][0].toHex(),
    })
    expect(entries2).toMatchSnapshot()
  })

  it('getKeysPagedAt', async () => {
    const apiAt = await api.at('0x6aa33cc6b9bfa9bbc3144597da05a1dfd3bf2d16fc52f67c1687fc4e7c8ced5f')

    const entries = await apiAt.query.system.account.entriesPaged({ args: [], pageSize: 10 })
    expect(entries).toMatchSnapshot()

    const entries2 = await apiAt.query.system.account.entriesPaged({
      args: [],
      pageSize: 10,
      startKey: entries[entries.length - 1][0].toHex(),
    })
    expect(entries2).toMatchSnapshot()
  })

  it('subscription', async () => {
    const { callback, next } = mockCallback()
    const unsub = await api.query.timestamp.now(callback)

    await next()

    expect(callback.mock.calls).toMatchSnapshot()
    callback.mockClear()

    expect(await dev.newBlock()).toMatchInlineSnapshot(
      '"0xadaa12ed69b66fa094b0d30c408b61dd944529981d9e902f15ed22ef3db54fed"'
    )

    await next()

    expect(callback.mock.calls).toMatchSnapshot()
    callback.mockClear()

    unsub()

    expect(await dev.newBlock()).toMatchInlineSnapshot(
      '"0x713d3e542b7ac7a4d446f10dc58b411ac6644ef1fd16f702941f4d7119116e17"'
    )

    await delay(100)

    expect(callback).not.toHaveBeenCalled()
  })
})
