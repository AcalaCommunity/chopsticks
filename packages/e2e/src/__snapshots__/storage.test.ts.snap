// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`storage > getKeysPaged 1`] = `
[
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9013adf9ce9b5fc252cd82afb5c8a6b9f65766d3a000000000000000000050000000000000000000e0000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da905f5da56ae126043f87dd8d86a593dea65766d3a00000000000000000005000000000000000000030000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90978ec581c3813694640edd550cea29465766d3a000000000000000000050000000000000000000b0000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9097df34545326553afe2cd1de615524e65766d3a00000000000000000001000000000000000000a80000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90a0cf4fe0af606730f032311099678ed65766d3a00000000000000000002000000008003000000020000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90d2f0036fcda4a149c85b40db3b3aab465766d3a00000000000000000000000000000000000008030000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90d55662592e78b28fe5a67e65d00e04165766d3a00000000000000000000000000000000000008010000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90e96e5d310327f39b4bc15041cdec6199e22b64c980329ada2b46a783623bcf1f1d0418f6a2b5fbfb7fb68dbac5abf0f",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": "0x00000000000000000de0b69ed8ad63ce",
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 941,
      "providers": 1,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90f0452ca76da89426565512fc5f338b265766d3a00000000000000000005000000000000000000130000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90f3d3e978061362057e94248f36c474465766d3a000000000000000000050000000000000000000a0000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
]
`;

exports[`storage > getKeysPaged 2`] = `
[
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90faf806ed89983038cb762cab52e40046d6f646c6163612f75726c730000000000000000000000000000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 1,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da91042bbb07d342995f4aa17576e606d7b8815a8024b06a5b4c8703418f52125c923f939a5c40a717f6ae3011ba7719019",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": "0x00000000000000000dbd2fbd2203a98b",
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 4,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9106e295f7efd2fbe873c49d8128748aa65766d3a00000000000000000000000000000000000008000000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da91241e3f2e98b1a98771830bbf2ee5ee188b3cb383e25bafa7195ab63bc677e1bd6cb27b97cd5d9e91e59f58bf1d3ea68",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": "0x0000000000000000001f9e693b0e3736",
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 4,
      "providers": 1,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9150054ea601c7ddef7fb45aeed0ba2b265766d3a00000000000000000002000000008403000000000000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da91c43c5fee0c875d839c6d4ac6bd55f8365766d3a00000000000000000000000000000000000008020000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da91cbd06372ac7c0113229c0901e32ed2965766d3a000000000000000000020000000002020000000d0000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da92136c00da386662d8c037011437b59356d6f646c6163612f6c6f616e0000000000000000000000000000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 1,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da92c33af6e20ceb120af213449fd67e5296d6f646c6163612f6465786d0000000000000000000000000000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 1,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da92e3ee09d462bb823b177689f68c802f865766d3abbbba9ebe50f9456e106e6ef29921791828899990000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 11000000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 1,
      "sufficients": 0,
    },
  ],
]
`;

exports[`storage > getKeysPagedAt 1`] = `
[
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9013adf9ce9b5fc252cd82afb5c8a6b9f65766d3a000000000000000000050000000000000000000e0000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da905f5da56ae126043f87dd8d86a593dea65766d3a00000000000000000005000000000000000000030000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90978ec581c3813694640edd550cea29465766d3a000000000000000000050000000000000000000b0000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9097df34545326553afe2cd1de615524e65766d3a00000000000000000001000000000000000000a80000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90a0cf4fe0af606730f032311099678ed65766d3a00000000000000000002000000008003000000020000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90d2f0036fcda4a149c85b40db3b3aab465766d3a00000000000000000000000000000000000008030000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90d55662592e78b28fe5a67e65d00e04165766d3a00000000000000000000000000000000000008010000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90e96e5d310327f39b4bc15041cdec6199e22b64c980329ada2b46a783623bcf1f1d0418f6a2b5fbfb7fb68dbac5abf0f",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": "0x00000000000000000de0b6b3a7640000",
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 1,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90f0452ca76da89426565512fc5f338b265766d3a00000000000000000005000000000000000000130000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90f3d3e978061362057e94248f36c474465766d3a000000000000000000050000000000000000000a0000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
]
`;

exports[`storage > getKeysPagedAt 2`] = `
[
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da90faf806ed89983038cb762cab52e40046d6f646c6163612f75726c730000000000000000000000000000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 1,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da91042bbb07d342995f4aa17576e606d7b8815a8024b06a5b4c8703418f52125c923f939a5c40a717f6ae3011ba7719019",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": "0x00000000000000000de0b6b3a7640000",
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9106e295f7efd2fbe873c49d8128748aa65766d3a00000000000000000000000000000000000008000000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9150054ea601c7ddef7fb45aeed0ba2b265766d3a00000000000000000002000000008403000000000000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da91c43c5fee0c875d839c6d4ac6bd55f8365766d3a00000000000000000000000000000000000008020000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da91cbd06372ac7c0113229c0901e32ed2965766d3a000000000000000000020000000002020000000d0000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da92136c00da386662d8c037011437b59356d6f646c6163612f6c6f616e0000000000000000000000000000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 1,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da92c33af6e20ceb120af213449fd67e5296d6f646c6163612f6465786d0000000000000000000000000000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 1,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da93621caf2585ed776e6229d3a9027e79565766d3a00000000000000000001000000000000000000820000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da938a28dd4a64b0cab7121447ed8568af765766d3a00000000000000000001000000000000000000840000000000000000",
    {
      "consumers": 0,
      "data": {
        "feeFrozen": 0,
        "free": 100000000000,
        "miscFrozen": 0,
        "reserved": 0,
      },
      "nonce": 0,
      "providers": 2,
      "sufficients": 0,
    },
  ],
]
`;

exports[`storage > getStorage 1`] = `1687283910060`;

exports[`storage > getStorage 2`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": "0x00000000000000000dbd2fbd2203a98b",
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 4,
  "providers": 2,
  "sufficients": 0,
}
`;

exports[`storage > getStorage 3`] = `1685410134036`;

exports[`storage > getStorage 4`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": "0x00000000000000000de0b6b3a7640000",
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 2,
  "sufficients": 0,
}
`;

exports[`storage > getStorageMulti 1`] = `
[
  {
    "consumers": 0,
    "data": {
      "feeFrozen": 0,
      "free": "0x00000000000000000dbd2fbd2203a98b",
      "miscFrozen": 0,
      "reserved": 0,
    },
    "nonce": 4,
    "providers": 2,
    "sufficients": 0,
  },
  {
    "consumers": 0,
    "data": {
      "feeFrozen": 0,
      "free": "0x00000000000000000de0b69ed8ad63ce",
      "miscFrozen": 0,
      "reserved": 0,
    },
    "nonce": 941,
    "providers": 1,
    "sufficients": 0,
  },
  {
    "consumers": 0,
    "data": {
      "feeFrozen": 0,
      "free": 100000000000,
      "miscFrozen": 0,
      "reserved": 0,
    },
    "nonce": 0,
    "providers": 2,
    "sufficients": 0,
  },
]
`;

exports[`storage > getStorageMulti 2`] = `
[
  {
    "consumers": 0,
    "data": {
      "feeFrozen": 0,
      "free": "0x00000000000000000de0b6b3a7640000",
      "miscFrozen": 0,
      "reserved": 0,
    },
    "nonce": 0,
    "providers": 2,
    "sufficients": 0,
  },
  {
    "consumers": 0,
    "data": {
      "feeFrozen": 0,
      "free": "0x00000000000000000de0b6b3a7640000",
      "miscFrozen": 0,
      "reserved": 0,
    },
    "nonce": 0,
    "providers": 1,
    "sufficients": 0,
  },
  {
    "consumers": 0,
    "data": {
      "feeFrozen": 0,
      "free": 100000000000,
      "miscFrozen": 0,
      "reserved": 0,
    },
    "nonce": 0,
    "providers": 2,
    "sufficients": 0,
  },
]
`;

exports[`storage > subscription 1`] = `
[
  [
    1687283910060,
  ],
]
`;

exports[`storage > subscription 3`] = `
[
  [
    1687283922060,
  ],
]
`;
